pkg/readers/adapters/amqp.go

```diff
@@ -24,12 +27,22 @@ func (a *Amqp) Listen(msgChannel chan<- messages.Message) {
 }
 
 func (a *Amqp) processMessages(writeChannel chan<- messages.Message) {
-	msg := messages.Message{}
-	for d := range a.ConsumedMessages {
-		msg.Id = d.DeliveryTag
-		msg.Body = d.Body
+	for {
+		select {
+		case err := <-a.Notify:
+			if err != nil {
+				log.Fatalf("Error in connection: %s", err)
```

Comment:
In case of error, log the error and exit. This should trigger the pod to restart and we delegate exponential backup to infrastructure, instead of implementing complex re-connection logic